lib/output.js

```diff
@@ -521,7 +522,17 @@ function OutputStream(options) {
     // a function expression needs parens around it when it's provably
     // the first token to appear in a statement.
     PARENS(AST_Function, function(output){
-        return first_in_statement(output);
+        if (first_in_statement(output)) {
+            return true;
+        }
+
+        if (output.option('wrap_iife')) {
+            var p = output.parent();
+            console.log()
+            return p instanceof AST_Call && p.expression === this;
```

Comment:
If #1310 is accurate then all function expressions are wrapped unconditionally.

If #1310 is incorrect then `negate_iife=false` should be sufficient and this option would not be required.